template/api/src/main/java/com/blazebit/notify/template/api/ServiceProvider.java

```diff
@@ -13,31 +13,24 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.blazebit.notify.email.model.memory;
+package com.blazebit.notify.template.api;
 
-import com.blazebit.notify.ConfigurationSourceProvider;
+import java.util.Optional;
 
 /**
- * A simple E-Mail notification.
+ * A service provider.
  *
- * @author Christian Beikov
+ * @author Moritz Becker
  * @since 1.0.0
  */
-public class EmailNotification extends AbstractEmailNotification<Long> implements ConfigurationSourceProvider {
-
-    private static final long serialVersionUID = 1L;
+public interface ServiceProvider {
```

Comment:
Why do we need a new `ServiceProvider` interface and can't just use the existing one?

Reply:
The existing one is part of blaze-job-core-api. So I would have had to introduce a dependency to blaze-job in order to reuse this class which I thought doesn't make sense (why would the template module depend on the job module).

Reply:
The template stuff is just an integration aspect of blaze-notify which already depends on the job module, so it's fine to add another dependency here IMO

Reply:
Why did we duplicate `com.blazebit.notify.template.api.ConfigurationSource` then instead of just using `com.blazebit.job.ConfigurationSource`?

Reply:
A mistake I guess.